pkg/providers/imagefamily/image.go

```diff
@@ -67,18 +74,23 @@ func (p *Provider) Get(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass, in
 	defaultImages := imageFamily.DefaultImages()
 	for _, defaultImage := range defaultImages {
 		if err := instanceType.Requirements.Compatible(defaultImage.Requirements, v1alpha2.AllowUndefinedWellKnownAndRestrictedLabels); err == nil {
-			communityImageName, publicGalleryURL := defaultImage.CommunityImage, defaultImage.PublicGalleryURL
-			imageID, err := p.GetImageID(ctx, communityImageName, publicGalleryURL)
-			if err != nil {
-				return "", "", err
-			}
-			return defaultImage.Distro, imageID, nil
+			imageID, imageRetrievalErr := p.GetLatestImageID(ctx, defaultImage)
+			return defaultImage.Distro, imageID, imageRetrievalErr
 		}
 	}
 
 	return "", "", fmt.Errorf("no compatible images found for instance type %s", instanceType.Name)
 }
 
+func (p *Provider) GetLatestImageID(ctx context.Context, defaultImage DefaultImageOutput) (string, error) {
+	// Managed Karpenter will use the AKS Managed Shared Image Galleries
+	if options.FromContext(ctx).UseSIG {
```

Comment:
Going forward, can we move away from provider(s) behaving differently based on options from context? I know this pattern is convinient, and already used in a couple of places, but it feels too coupled to context, harder to test, etc. I'd much rather see the required configuration flowing into provider instantiation. *Does not have to be addressed now.)